atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ConjureJavaRuntimeTargetFactory.java

```diff
@@ -76,19 +76,16 @@ private ConjureJavaRuntimeTargetFactory() {
 
     @Override
     public <T> InstanceAndVersion<T> createLiveReloadingProxyWithFailover(
-            Supplier<ServerListConfig> serverListConfigSupplier,
+            Refreshable<ServerListConfig> serverListConfigRefreshable,
             Class<T> type,
             AuxiliaryRemotingParameters parameters) {
         ClientOptions options = getClientOptionsForFailoverProxy(parameters);
-        Supplier<T> clientSupplier = new CachedTransformingSupplier<>(
-                serverListConfigSupplier,
-                serverListConfig -> JaxRsClient.create(
-                        type,
-                        addAtlasDbRemotingAgent(parameters.userAgent()),
-                        NoOpHostEventsSink.INSTANCE,
-                        options.serverListToClient(serverListConfig)));
-
-        return decorateFailoverProxy(type, clientSupplier);
+        Refreshable<T> refreshableClient = serverListConfigRefreshable.map(serverListConfig -> JaxRsClient.create(
+                type,
+                addAtlasDbRemotingAgent(parameters.userAgent()),
+                NoOpHostEventsSink.INSTANCE,
+                options.serverListToClient(serverListConfig)));
+        return decorateFailoverProxy(type, refreshableClient);
     }
 
     public <T> InstanceAndVersion<T> createProxyWithQuickFailoverForTesting(
```

Comment:
this method is not being used?

Reply:
I didn't want to break the public API, but given the name and fact that it is not used in large internal product, should be fine